### ISSUE TITLE
Pathspider not setting destination port fallback value for TCP

### ISSUE THREAD

**Issue by nstudach** [mami-project/pathspider#240]
When measuring with pathspider and the MSS plugin the following error occurs:
```
INFO:pathspider:activating spider...
INFO:pathspider:starting pathspider
INFO:pathspider:Creating observer
INFO:desync:configurations are not synchronized
INFO:pathspider:opening output file 7S415-blr1-mss
ERROR:pathspider:exception occurred. terminating.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pathspider-2.0.0.dev0-py3.6.egg/pathspider/base.py", $
    target(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pathspider-2.0.0.dev0-py3.6.egg/pathspider/desync.py"$
    job, config, connect=self.connections[config])
  File "/usr/local/lib/python3.6/dist-packages/pathspider-2.0.0.dev0-py3.6.egg/pathspider/base.py", $
    conn = connect(job, config)
  File "/usr/local/lib/python3.6/dist-packages/pathspider-2.0.0.dev0-py3.6.egg/pathspider/sync.py", $
    rec = connect_tcp(self.source, job, self.args.timeout)
  File "/usr/local/lib/python3.6/dist-packages/pathspider-2.0.0.dev0-py3.6.egg/pathspider/helpers/tc$
    sock.connect((job['dip'], job['dp']))
KeyError: 'dp'
INFO:pathspider:terminating pathspider
```

# Input File:
```
{"dip":"46.245.167.89","domain":"mavikadin.com","hellfire_lookup_attempts":1,"hellfire_lookup_type":"host","rank":"25192"}
{"dip":"2400:cb00:2048:1::6814:cb1c","domain":"samnytt.se","hellfire_lookup_attempts":1,"hellfire_lookup_type":"host","rank":"25209"}
{"dip":"2400:cb00:2048:1::6814:ca1c","domain":"samnytt.se","hellfire_lookup_attempts":1,"hellfire_lookup_type":"host","rank":"25209"}
{"dip":"104.20.202.28","domain":"samnytt.se","hellfire_lookup_attempts":1,"hellfire_lookup_type":"host","rank":"25209"}
{"dip":"104.20.203.28","domain":"samnytt.se","hellfire_lookup_attempts":1,"hellfire_lookup_type":"host","rank":"25209"}
{"dip":"2400:cb00:2048:1::6810:1305","domain":"cropp.com","hellfire_lookup_attempts":1,"hellfire_lookup_type":"host","rank":"25208"}
```

when adding the following code to pathspider/helpers/tcp.py: in the function `def connect_tcp(source, job, conn_timeout, sockopts=None):`

```
    if 'dp' not in job:
        job['dp'] = 80
```

It terminates as expected.

# Solution:
Add port  fallback value for tcp.py

I do not know what the appropriate port would be. Can someone clear this up for me?

Thanks,
Noah

**Comment by irl** [comment]
The expected workflow is to use `pspdr filter` to add a destination port to your resolved target lists.

Throwing a `KeyError` is definitely a bug though. Either we should make this a warning and carry on (maybe only one entry in an otherwise good list is missing a destination port) or we should make this fatal and point the user at `pspdr filter` or we should add the fallback to port 80.

If you set this in the `connect_tcp` function does this get copied into the output? I think it might not because that job record is just a stub. The full job record has already been sent to the combiner at that stage. This would mean that a fallback has been used but not recorded in the output, and this harms comparability/reproducibility.

**Comment by nstudach** [comment]
the port does get carried over:
```
{"dip": "104.20.202.28", "domain": "samnytt.se", "hellfire_lookup_attempts": 1, "hellfire_lookup_type": "host", "rank": "25209", "sip": "172.17.18.186", "path": ["172.17.18.186", "217.110.71.114", "AS8220", "104.20.202.28"], "sip_public": "217.110.71.114", "sip_asn": 8220, "dp": 80, "time": {"from": "2018-08-25 19:43:45.983663", "to": "2018-08-25 19:43:45.993930"}, "conditions": ["mss.connectivity.online", "mss.option.local.value:1460", "mss.option.remote.value:1460", "mss.option.received.unchanged"]}
```

**Comment by irl** [comment]
Ok then. Can you make a pull request to add that fallback then? That's probably the easiest option.

**Comment by irl** [comment]
This issue is still unresolved. Keeping issue open until it is fixed.

@nstudach Ok. If I get to it first then I will do it but I am also away until the 7th.

**Comment by irl** [comment]
Fixed in abc688c. Thanks @nstudach!
